ethers-providers/src/provider.rs

```diff
@@ -786,6 +789,145 @@ impl<P: JsonRpcClient> Middleware for Provider<P> {
         self.query_resolver(ParamType::String, &ens_name, ens::NAME_SELECTOR).await
     }
 
+    /// Returns the avatar HTTP link of the avatar that the `ens_name` resolves to (or None
+    /// if not configured)
+    ///
+    /// # Example
+    /// ```
```

Comment:
let's do no_run here
```suggestion
    /// ```no_run
```